pkg/cloud_provider/clientset/clientset.go

```diff
@@ -43,9 +45,10 @@ type PodInfo struct {
 
 type Clientset struct {
 	k8sClients kubernetes.Interface
+	podLister  listersv1.PodLister
 }
 
-func New(kubeconfigPath string) (Interface, error) {
+func New(kubeconfigPath, nodeName string) (Interface, error) {
 	var err error
 	var rc *rest.Config
 	if kubeconfigPath != "" {
```

Comment:
should we also return an error  if node name is an empty string?

also we already fatal in L78 in main.go, so we can return error from here

Reply:
If we return an error when the node name is empty, it will always crash in the CSI controller because controller does not pass a node name, although the CSI controller is not in production.

I refactored the code to only prepare the pod informer in CSI Node server. CSI controller does not use pod informer at all.